eslint-plugin-prettier.js

```diff
@@ -0,0 +1,306 @@
+  const pos = getLocFromIndex(context, offset);
+  const range = [null, offset];
+  context.report({
+    message: 'Insert `{{ code }}`',
+    data: { code: showInvisibles(text) },
+    loc: { start: pos, end: pos },
+    fix(fixer) {
+      return fixer.insertTextAfterRange(range, text);
+    }
+  });
+}
+
+/**
+ * Reports a "Delete ..." issue where text must be deleted.
+ * @param {RuleContext} context - The ESLint rule context.
+ * @param {number} offset - The source offset where to delete text.
+ * @param {string} text - The text to be deleted.
+ * @returns {void}
+ */
+function reportDelete(context, offset, text) {
+  const start = getLocFromIndex(context, offset);
+  const end = getLocFromIndex(context, offset + text.length);
+  const range = [offset, offset + text.length];
+  context.report({
+    message: 'Delete `{{ code }}`',
+    data: { code: showInvisibles(text) },
+    loc: { start, end },
+    fix(fixer) {
+      return fixer.removeRange(range);
+    }
+  });
+}
+
+/**
+ * Reports a "Replace ... with ..." issue where text must be replaced.
+ * @param {RuleContext} context - The ESLint rule context.
+ * @param {number} offset - The source offset where to replace deleted text
+ with inserted text.
+ * @param {string} deleteText - The text to be deleted.
+ * @param {string} insertText - The text to be inserted.
+ * @returns {void}
+ */
+function reportReplace(context, offset, deleteText, insertText) {
+  const start = getLocFromIndex(context, offset);
+  const end = getLocFromIndex(context, offset + deleteText.length);
+  const range = [offset, offset + deleteText.length];
+  context.report({
+    message: 'Replace `{{ deleteCode }}` with `{{ insertCode }}`',
+    data: {
+      deleteCode: showInvisibles(deleteText),
+      insertCode: showInvisibles(insertText)
+    },
+    loc: { start, end },
+    fix(fixer) {
+      return fixer.replaceTextRange(range, insertText);
+    }
+  });
+}
+
+// ------------------------------------------------------------------------------
+//  Module Definition
+// ------------------------------------------------------------------------------
+
+module.exports.rules = {
+  prettier: {
+    meta: {
+      fixable: 'code',
+      schema: [
+        // Prettier options:
+        {
+          anyOf: [
+            { enum: [null, 'fb'] },
+            { type: 'object', properties: {}, additionalProperties: true }
+          ]
+        },
+        // Pragma:
+        { type: 'string', pattern: '^@\\w+$' }
+      ]
+    },
+    create(context) {
+      const prettierOptions = context.options[0] === 'fb'
+        ? FB_PRETTIER_OPTIONS
+        : context.options[0];
+
+      const pragma = context.options[1]
+        ? context.options[1].slice(1) // Remove leading @
+        : null;
+
+      if (pragma) {
+        // The pragma is only valid if it is found in a block comment at the
+        // very start of the file.
+        const firstComment = context.getAllComments()[0];
```

Comment:
I think `context.getAllComments()` is also deprecated. Can you use `context.getSourceCode().getAllComments()` instead?

Also, starting in ESLint 4, shebang comments will be included in the result of `sourceCode.getAllComments` (see the [migration guide](http://eslint.org/docs/4.0.0/user-guide/migrating-to-4.0.0#shebangs)). Shebang comments need to be placed at the start of a file, so maybe it would be better to filter them here.